include/circt/Dialect/ESI/ESIManifest.td

```diff
@@ -49,6 +49,10 @@ def AppIDPathAttr : ESI_Attr<"AppIDPath"> {
   let assemblyFormat = [{
     `<` $root `[` $path `]` `>`
   }];
+
+  let extraClassDeclaration = [{
+    AppIDPathAttr getParent();
```

Comment:
missing comment - is it the parent operation? the next level up in the path? ...